src/library_gl.js

```diff
@@ -986,7 +986,10 @@ var LibraryGL = {
         var u = GLctx.getActiveUniform(p, i);
 
         var name = u.name;
-        ptable.maxUniformLength = Math.max(ptable.maxUniformLength, name.length+1);
+        var length = name.length + 1;
+        if (length > ptable.maxUniformLength) {
+          ptable.maxUniformLength = length;
+        }
```

Comment:
why is this change shorter? just because Math.max does not minify down in closure?

if that's the issue, I'd rather not do this, as it decreases readability. Instead closure, or we, could do `var x = Math.max` once at the toplevel, and we'd use the minified name here.

Reply:
That's right, `Math.max` does not minify since it is an extern. Without Math.max the code becomes
```js
var h=L.getActiveUniform(b,d),k=h.name,l=k.length+1;l>a.L&&(a.L=l);l=k.lastIndexOf("[")
```
whereas with `Math.max()` code is
```js
var h=L.getActiveUniform(b,d),k=h.name;a.L=Math.max(a.L,k.length+1);var l=k.lastIndexOf("[");
```
Closure has a bug there it looks like, it could have done
```js
var h=L.getActiveUniform(b,d),k=h.name,l;a.L=Math.max(a.L,k.length+1);l=k.lastIndexOf("[");
```
to save 2 more bytes, but that would still be 4 bytes longer than the version without Math.max.


Also,
```js
var x=Math.max;p.m=x(p.m,l);
```
would be longer than just
```js
p.m=Math.max(p.m,l);
```
so the assignment `var x=Math.max;` is only shorter if it can be amortized between multiple uses of `Math.max`. Searching candidates to amortize with shows only `emscripten_set_timeout_loop()` in asm.js mode as worthwhile, but not many apps would necessarily use that.

Code pattern
```js
var length = ...;
if (length > max) {
  max = length;
}
```
does not look too unusual, it is after all what every beginner programmer writes when they start out and are not yet familiar with the language's built-in math primitives. Closure is just telling they are not wrong :)

Reply:
What about `Math_max`, though? with a `_` there, closure should minify it properly? We do already define that, so you can just use it.

Reply:
`var Math_max=Math.max;` will minify down to `var x=Math.max;` from above. The define does not come free, it is not getting used anywhere, and it's getting optimized out. If it was not optimized out, it still would need to amortize for it to warrant its weight.

Reply:
It's true we'd still have `var x=Math.max`, but that's a single cost compared to `if(a<b)a=b;` in multiple places (which we'd replace with `a=x(a,b);`. So with enough uses that's better. I don't think we should bet against not having more uses.

If we do want to super-optimize stuff like this, see my other comment about the new estree optimizer - we could write a specific pass to optimize this exact thing, to see how many max operations there are, and pick between the two forms globally.

Reply:
Hmm, I think that would be a complete overkill in terms of complexity.. It should be much simpler to have a `if (x > max) max = x;` in code, compared to writing a full optimization pass to do the job?